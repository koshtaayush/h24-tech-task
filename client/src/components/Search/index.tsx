//Library
import React from 'react'
import styled from 'styled-components'

//Resuable Components
import Input from './../Input'

interface Props {
    placeholder: string
    value: string
    onChangeProp: (value: string) => void
}

const Search: React.FC<Props> = (props) => {

    const { placeholder, value, onChangeProp } = props;

    return (
        <React.Fragment>
            <TopSection>
                <InputContainer>
                    <Input
                        onChangeProp={onChangeProp}
                        inputValue={value}
                        placeHolderText={placeholder}
                        test-id="searchInput"
                    />
                </InputContainer>
            </TopSection>
        </React.Fragment>
    )
}

export default Search

const TopSection = styled.div`
`

const InputContainer = styled.div`
    position: relative;
`
