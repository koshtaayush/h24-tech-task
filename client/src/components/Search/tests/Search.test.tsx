import React from 'react'
import { shallow } from 'enzyme'

import Search from '..'

describe('Search', () => {
    it('Should match the snapshot', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Search
                placeholder={'Placeholder text'}
                value= {'Input Value'}
                onChangeProp= {mockFn}
            />
        )

        expect(component).toMatchSnapshot()
    })
})
