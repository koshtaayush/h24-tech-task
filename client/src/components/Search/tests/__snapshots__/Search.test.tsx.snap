// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should match the snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Input
        inputValue="Input Value"
        onChangeProp={[MockFunction]}
        placeHolderText="Placeholder text"
        test-id="searchInput"
      />
    </styled.div>
  </styled.div>
</Fragment>
`;
